src/bjointsp/read_write/writer.py

```diff
@@ -90,9 +90,9 @@ def save_heuristic_variables(result, changed_instances, instances, edges, nodes,
             result["placement"]["flows"].append(flow)
         for path in e.paths:
             # record edge delay: all flows take the same (shortest) path => take path delay
-            delay = {"src": e.arc.source.name, "dest": e.arc.dest.name, "src_node": e.source.location, "dest_node": e.dest.location, "delay": sp.path_delay(links, path)}
+            delay = {"src": e.arc.source.name, "dest": e.arc.dest.name, "src_node": e.source.location, "dest_node": e.dest.location, "path_delay": sp.path_delay(links, path), "src_vnf_delay": e.arc.source.vnf_delay, "dest_vnf_delay": e.arc.dest.vnf_delay }
             result["metrics"]["delays"].append(delay)
-            result["metrics"]["total_delay"] += sp.path_delay(links, path)
+            result["metrics"]["total_delay"] += sp.path_delay(links, path)+e.arc.source.vnf_delay+e.arc.dest.vnf_delay
```

Comment:
This is wrong because it will add the vnf processing delays multiple times. Eg, in a chain of A->B->C, the VNF delay of B is added as dest_vnf in A->B and as src vnf in B->C.

Please just add vnf processing delays in a separate loop and don't mix it with the link delays. that's cleaner anyways